saleor/graphql/utils.py

```diff
@@ -20,6 +24,49 @@ def get_database_id(info, node_id, only_type):
     return _id
 
 
+def _check_graphene_type(requested_graphene_type, received_type):
+    if requested_graphene_type:
+        assert str(requested_graphene_type) == received_type, (
+            'Must receive an {} id.'
+        ).format(requested_graphene_type._meta.name)
+
+
+def _resolve_nodes(ids, graphene_type=None):
+    pks = []
+    types = []
+    invalid_ids = []
+
+    for graphql_id in ids:
+        if not graphql_id:
+            continue
+
+        try:
+            _type, _id = from_global_id(graphql_id)
+        except Exception:
+            invalid_ids.append(graphql_id)
+            continue
+
+        _check_graphene_type(graphene_type, _type)
+        pks.append(_id)
+        types.append(_type)
```

Comment:
@maarcingebala I spoke with @dominik-zeglen and both of us can't recall a case when client uses list with items of different types. This list seems to be legacy of idea that someone one day would.
Do you think we can drop it in favor of something like
```
# initally
type = None
# in the loop
_type, _id = from_global_id(graphql_id)
if _type is None:
    type = _type
else:
    assert _type == type, 'Received IDs of more than one type.' # Maybe raise?
```
It is not a big deal but we could avoid maintaining list that we expect to be one element one anyway.
Unless of course there is a case, when we want to use more types in one query.
What do you think?

Reply:
I agree - this function's purpose should be returning objects of a single type; the presence of a type different than the expected one should be considered an error.

Reply:
Let me know if that was what you were thinking https://github.com/mirumee/saleor/pull/3942/commits/e147063275686d481113db040c51648f234f6188

Reply:
Yes, it's ok!